### ISSUE TITLE
Accessibility id is not visible for ios simulator assigned to textInput in appium2 react native

### ISSUE THREAD

**Issue by chetnaag95** [appium/appium#18642]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Hi,

Accessibility ID is not visible in appium inspector on inspecting textInput in react native for ios simulator, but are visible on Button 

Also the same accessible id are visible for android in appium inspector

**Code snippet** - 
<TextInput
            accessible
            accessibilityLabel={translateLabel(this.translate, 'general', 'field', 'email_phone')}
            accessibilityHint="Navigates to the previous screen"
            editable={XXX}
            style={[this.state.style.username.field, alignStyle, { margin: 5 }]}
            placeholder={this.translate('email_phone')}
            value={this.state.username}
            autoCapitalize="none"
            autoCorrect={false}
            spellCheck={false}
            dataDetectorTypes="none"
            placeholderTextColor={GREY}
            underlineColorAndroid="transparent"
          />

Appium - v2.0.0-beta.66
WDA - v4.9.0
Node - v14.21.3
MacOS - Ventura 13.2
Xcode - 14.2
npm - 9.6.6

### Expected Behavior

Accessbility label attribute is already added in textInput as I added in Button. 

accessibility id should reflect in appium inspector for ios simulator as it appears for android

### Minimal Reproducible Example

accessibility id should reflect in appium inspector for ios simulator as it appears for android

### Environment

- Operating system:
- If running via `appium` CLI...
  - Appium CLI version (output of `appium --version`):
  - Node.js version (output of `node --version`):
  - `npm` version (output of `npm --version`): 
  - Last `appium` version which did _not_ exhibit the problem: 
- If running Appium Desktop, its version:
- Appium driver(s) and their version(s):
- Appium plugin(s) and their version(s):
- Platform and version under test:
- Real device or emulator/simulator:

**Comment by mykola-mokhnach** [comment]
XCTest only sees what is exposed to accessibility. Please check the react tutorial on accessibility interaction. There is nothing that could be done from Appium side to change that behaviour

**Comment by chetnaag95** [comment]
@mykola-mokhnach I can see the accessibility label if I check it with xcode accessibility inspector

**Comment by mocnidule** [comment]
it is actually on the Appium side. It struggles to find nested elements, even if they were given datatest-id or accessibility-id. Here is an example below for an element that just can't be found using Appium, regardless of snapshotMaxDepth set.

The first screenshot is Appium inspector, the second one is Accessibility inspector. With Accessibility inspector it is found, no problem. But Appium ... 


<img width="1164" alt="Screenshot 2024-05-15 at 3 54 01 PM" src="https://github.com/appium/appium/assets/58424871/323135e7-0d62-4b33-a3b6-c3b3601a3a3f">


<img width="1653" alt="Screenshot 2024-05-15 at 3 53 43 PM" src="https://github.com/appium/appium/assets/58424871/384879b1-bbf7-43ef-a4c5-b8a9a48d457a">

**Comment by KazuCocoa** [comment]
What about via a real device, and [`"mobile:source"` with `"description" format](https://appium.github.io/appium-xcuitest-driver/latest/reference/execute-methods/#mobile-source)? The accessibility inspector and XCTest framework could show different info. Simulator could be part of host machine for accessibility info.

Btw, what about this section? Perhaps current selecting place is center of the cell?
<img width="39" alt="Screenshot 2024-05-15 at 8 12 59 AM" src="https://github.com/appium/appium/assets/5511591/568686ba-8dcd-42b5-85ce-b56f0b9c99e4">

**Comment by mocnidule** [comment]
Hey @KazuCocoa, thanks for the reply. I haven't yet tested with a real device. 

Can you, when you find time, elaborate more on:

> The accessibility inspector and XCTest framework could show different info. Simulator could be part of host machine for accessibility info.

I am interested why they would show different info

The section, if you are referring to the square is the vertical scroll.

**Comment by KazuCocoa** [comment]
> Simulator could be part of host machine for accessibility info.

This might be wrong. In accessibility inspector on macOS 13.6.6 showed a simulator process separately from the host machine. Maybe... old inspector included simulator as part of host machine's process.

> I am interested why they would show different info

It's black-box so cannot say 100% sure thing but I saw XCTest (not via Appium) and accessibility inspector had different element findings, this was my very old experience, so might be different from current behavior. WDA depends on https://developer.apple.com/documentation/xctest/user_interface_tests . Accessibility inspector may not depend on XCTest but XCTest probably depends on the accessibility inspector's info. So this is basically observation basis guess.

You could try enabling `Settings -> Accessibility -> Spoken Content -> Speak Selection` in https://appium.github.io/appium-xcuitest-driver/latest/preparation/#accessibility-settings. I experienced it could expose more elements in Flutter/React Native.
